masonry_winit/src/event_loop_runner.rs

```diff
@@ -300,66 +291,43 @@ impl MasonryState<'_> {
 
     // --- MARK: RESUMED
     pub fn handle_resumed(&mut self, event_loop: &ActiveEventLoop, app_driver: &mut dyn AppDriver) {
+        self.app_state = AppState::Resumed;
+
+        // Create initial windows.
         if !self.initial_windows.is_empty() {
             for (id, attrs, widget) in std::mem::take(&mut self.initial_windows) {
                 self.create_window(event_loop, id, attrs, widget);
             }
-        } else {
-            for mut window in std::mem::take(&mut self.windows).into_values() {
-                match std::mem::replace(
-                    &mut window.state,
-                    // TODO: Is there a better default value which could be used?
-                    WindowState::Uninitialized(WindowAttributes::default()),
-                ) {
-                    WindowState::Uninitialized(attributes) => {
-                        self.create_window_inner(event_loop, attributes, window);
-                    }
-                    WindowState::Suspended {
-                        handle,
-                        accesskit_adapter,
-                    } => {
-                        window.state = WindowState::Rendering {
-                            handle,
-                            accesskit_adapter,
-                        };
-                    }
-                    _ => {
-                        // We have received a redundant resumed event. That's allowed by winit
-                    }
-                }
-            }
+        }
+
+        //  Recreate surfaces for all windows.
+        for (handle_id, window) in self.windows.iter() {
+            let surface = create_surface(&mut self.render_cx, window.handle.clone());
+            self.surfaces.insert(*handle_id, surface);
         }
```

Comment:
I guess it isn't documented that creating a surface twice for the same window is a problem. It certainly feels fragile, though, and so I'd like to change it.

Reply:
I see, I'm not quite familiar with the constraints, is there any pointers on how to do that? Thanks!

Reply:
I see, but we did short-circuit the function isn't it to prevent the recreation of the surfaces?

Reply:
I think my latest push fixes it. We will re-create all the surfaces before creating new windows to prevent creating the same surface twice.